packages/@aws-cdk/aws-codecommit/lib/pipeline-action.ts

```diff
@@ -54,11 +54,15 @@ export class PipelineSourceAction extends codepipeline.SourceAction {
       configuration: {
         RepositoryName: props.repository.repositoryName,
         BranchName: props.branch || 'master',
-        PollForSourceChanges: props.pollForSourceChanges !== undefined ? props.pollForSourceChanges : true
+        PollForSourceChanges: props.pollForSourceChanges || false,
       },
       outputArtifactName: props.outputArtifactName
     });
 
+    if (props.pollForSourceChanges !== true) {
+      props.repository.onCommit('CodePipeline', props.stage.pipelineAsEventTarget, props.branch || 'master');
```

Comment:
What happens if we want multiple pipelines to subscribe to the same repo? I think your ID should include the pipeline's `.uniqueId` or something.

Reply:
Good call. It's actually a miss on my part, as I meant to change this name to something else, but forgot. Will add the Pipeline's `uniqueId`.